src/node_http2.cc

```diff
@@ -69,8 +69,55 @@ Http2Options::Http2Options(Environment* env) {
   }
 }
 
-void Http2Session::OnFreeSession() {
-  ::delete this;
+
+Http2Session::Http2Session(Environment* env,
+                           Local<Object> wrap,
+                           nghttp2_session_type type)
+    : AsyncWrap(env, wrap, AsyncWrap::PROVIDER_HTTP2SESSION),
+      StreamBase(env) {
+  MakeWeak<Http2Session>(this);
+
+  Http2Options opts(env);
+
+  padding_strategy_ = opts.GetPaddingStrategy();
+
+  Init(type, *opts);
+
+  // For every node::Http2Session instance, there is a uv_prepare_t handle
+  // whose callback is triggered on every tick of the event loop. When
+  // run, nghttp2 is prompted to send any queued data it may have stored.
+  prep_ = new uv_prepare_t();
```

Comment:
btw, definitely not for this PR, but down the road... I plan to refactor this so that there is *one* uv_prepare_t for N session instances, and loop through each on every uv_prepare_t tick... rather than registering one uv_prepare_t for each session. Will need to benchmark that tho...

Reply:
:+1: – I can leave a `TODO` if you want.

I’ve also wondered if it might worth hopping onto the same underlying mechanism that `setImmediate` uses…

Reply:
nah, don't need a todo. 

> ... the same underlying mechanism that `setImmediate` uses...

I've considered that also but haven't looked into it yet.